### ISSUE TITLE
Visual Studio Compile on Save not immediately wiring up after rename

### ISSUE THREAD

**Issue by nycdotnet** [Microsoft/TypeScript#1768]
The Compile on Save functionality in Visual Studio 2013 when initially converting a file to TypeScript from JavaScript is buggy.  This may result in a negative first impression of TypeScript.

**Repro:**
  * Brand new Windows 8.1 fully patched with VS 2013 Community Edition (Update 4). No extensions or other software installed.
  * Start a new C# ASP.NET Web Application
  * Select the "Empty" template, uncheck "Host in the cloud".
  * Right-click Add JavaScript file.  Call it `test.js`.
  * Enter some simple code such as:

```javascript
var x = 1;
x = x + 10;
console.log(x);
```

  * Save it.
  * Right-click rename file to `test.ts`.  Answer yes to "change extension" warning, No to "Seach NuGet?" question.
  * Save the file.  **Expected:** Since there are no errors and "compile on save" is enabled, Visual Studio should compile the TypeScript file and generate a source map.  **Actual:** The file is not compiled and no source maps are generated.
  * Make a trivial change to the file (add space or change numbers that are being added).  Try saving again.  **Result:** Still no compile.
  * Close the TypeScript file and open it again.
  * Click save without making any changes.  **Result:** Still no compile.
  * Make a trivial change and save it.  **Result:** Still no compile.
  * Right-click "Unload project", then Right-click "Reload project".
  * Make another trivial change and save it.  **Result:** TypeScript file compiles as expected.
  * Right-click Add JavaScript file.  Call it `test2.js`.
  * Put in some simple code such as:

```javascript
var y = 9;
```

  * Right-click rename file to `test2.ts`.
  * Save `test2.ts`.  **Result:** test2.js and test2.map.js are not emitted.
  * Make a trivial change to `test.ts` (the original file) and save it.  **Result:** test.js and test.js.map are emitted as expected, so only `test2` is broken.

This also occurs if TypeScript 1.4 has been installed from the Visual Studio Gallery.

**Comment by danquirk** [comment]
I suspect renaming a .js to .ts does not trigger the same auto configuration of the project for TypeScript  as Add New Item -> TypeScript file.

**Comment by nycdotnet** [comment]
Yes, I wouldn't be surprised if this and #1759 had a similar root cause.  Thanks for looking into this, Dan.

**Comment by mhegazy** [comment]
@nycdotnet this should be fixed in VS 2015. We have rewrote most of the compile-on-save implementation for this release to handle this class of issues. Porting the fix back to VS2013 would not be farly expensive. Closing this as won't fix; please reopen if you see issues on VS2015.

**Comment by nycdotnet** [comment]
Did you mean would be fairly expensive or would not be?  As of now, do you intend to backport into VS 2013 update 5?  If I update my TypeScript course, I can include that info.

Thanks all - I look forward to trying it in VS 2015 when it RTMs in a few weeks.  :-)

-Steve O

**Comment by mhegazy** [comment]
Back porting the fixes to VS2013 would be expensive; we had to do some architectural changes to make sure this class of issues are addressed. So no change coming in Update5 for this issue in VS2013.

**Comment by nycdotnet** [comment]
Thanks so much.  I can't wait for 1.5 official, not to mention VS 2015 RTM.
